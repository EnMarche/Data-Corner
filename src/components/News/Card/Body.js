import { Grid, Button } from '@mui/material'
import { makeStyles } from '@mui/styles'
import NewsEnableStatus from './NewsEnableStatus'
import PropTypes from 'prop-types'
import News from 'domain/news'

const messages = {
  see: 'Voir',
}

const useStyles = makeStyles(theme => ({
  buttonContainer: {
    position: 'relative',
    bottom: '0',
  },
  editButton: {
    fontSize: '13px',
    fontWeight: '500',
    color: theme.palette.orange500,
    marginTop: theme.spacing(1.5),
    '&:hover': {
      background: theme.palette.newsBackground,
      borderRadius: '8.35px',
    },
  },
}))

const Body = ({ news, handleClickOpen, toggleStatus }) => {
  const classes = useStyles()
  const { id, status } = news

  return (
    <>
      <Grid container justifyContent="space-between" className={classes.buttonContainer}>
        <Grid item>
          <Button className={classes.editButton} onClick={() => handleClickOpen(id)}>
            {messages.see}
          </Button>
        </Grid>
        <Grid item>
          <NewsEnableStatus id={id} status={status} toggleStatus={toggleStatus} />
        </Grid>
      </Grid>
    </>
  )
}

Body.propTypes = {
  news: News.propTypes.isRequired,
  handleClickOpen: PropTypes.func.isRequired,
  toggleStatus: PropTypes.func.isRequired,
}

export default Body
