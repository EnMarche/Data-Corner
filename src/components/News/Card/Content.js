import { Grid, Button as MuiButton } from '@mui/material'
import { styled } from '@mui/system'
import NewsStatus from './NewsStatus'
import PropTypes from 'prop-types'
import News from 'domain/news'

const Button = styled(MuiButton)(
  ({ theme }) => `
  font-size: 13px;
  font-weight: 500;
  color: ${theme.palette.orange500};
  margin-top: ${theme.spacing(1.5)};
  &:hover {
    background: ${theme.palette.newsBackground};
    border-radius: 8.35px;
  };
`
)

const messages = {
  see: 'Voir',
}

const Content = ({ news, handleClick, toggleStatus }) => {
  const { id, status } = news

  return (
    <>
      <Grid container justifyContent="space-between">
        <Grid item>
          <Button onClick={handleClick}>{messages.see}</Button>
        </Grid>
        <Grid item>
          <NewsStatus id={id} status={status} toggleStatus={toggleStatus} />
        </Grid>
      </Grid>
    </>
  )
}

Content.propTypes = {
  news: News.propTypes.isRequired,
  handleClick: PropTypes.func.isRequired,
  toggleStatus: PropTypes.func.isRequired,
}

export default Content
