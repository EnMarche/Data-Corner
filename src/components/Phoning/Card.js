import PropTypes from 'prop-types'
import { Card, CardActions, CardContent, CardHeader } from '@mui/material'

const UICard = ({
  headerTitle,
  headerSubtitle,
  content,
  actions,
  rootProps = {},
  headerProps = {},
  contentProps = {},
  actionsProps = {},
}) => (
  <Card {...rootProps}>
    {headerTitle && <CardHeader {...headerProps} title={headerTitle} subheader={headerSubtitle} />}

    <CardContent sx={{ ...(contentProps.sx || {}), py: 0 }} {...contentProps}>
      {content}
    </CardContent>

    {actions && <CardActions {...actionsProps}>{actions}</CardActions>}
  </Card>
)

UICard.propTypes = {
  headerTitle: PropTypes.node,
  headerSubtitle: PropTypes.node,
  content: PropTypes.node,
  actions: PropTypes.node,
  rootProps: PropTypes.object,
  headerProps: PropTypes.object,
  contentProps: PropTypes.object,
  actionsProps: PropTypes.object,
}

export default UICard
