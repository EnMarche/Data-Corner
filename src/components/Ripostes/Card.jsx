/* eslint-disable react/forbid-prop-types */
import React from 'react';
import {
    makeStyles, createStyles, Grid, Paper, Button, Box,
} from '@material-ui/core';
import PropTypes from 'prop-types';
import DotMenu from '../HelperComponents/DotMenu';

const useStyles = makeStyles((theme) => createStyles({
    root: {
        padding: '16px',
        borderRadius: '8.35px',
    },
    container: {
        marginBottom: '8px',
    },
    chip: {
        fontSize: '10px',
        fontWeight: '500',
        borderRadius: '19px',
        padding: '2px 8px',
    },
    withBorder: {
        border: `1px solid ${theme.palette.gray200}`,
    },
    active: {
        color: theme.palette.teal700,
        background: theme.palette.activeLabel,
        borderRadius: '19px',

    },
    inactive: {
        color: theme.palette.red600,
        background: theme.palette.inactiveLabel,
    },
    date: {
        fontSize: '10px',
        color: theme.palette.gray600,
        padding: '7px 8px',
    },
    title: {
        fontSize: '16px',
        fontWeight: '600',
        color: theme.palette.gray900,
        width: '400px',
        whiteSpace: 'nowrap',
        overflow: 'hidden',
        textOverflow: 'ellipsis',
    },
    creator: {
        fontSize: '12px',
        fontWeight: '400',
        color: theme.palette.gray600,
    },
    KpiContainer: {
        marginBottom: '20px',
    },
    buttonContainer: {
        position: 'relative',
        bottom: '0',
    },
    editButton: {
        fontSize: '13px',
        fontWeight: '500',
        color: theme.palette.teal700,
        marginTop: '11px',
        '&:hover': {
            background: theme.palette.riposteBackground,
            borderRadius: '8.35px',
        },
    },
}));

const Card = ({
    item, handleClickOpen, handleActiveItem,
}) => {
    const classes = useStyles();

    return (
        <Grid item xs={12} sm={6} md={4}>
            <Paper classes={{ root: classes.root }}>
                <Grid container className={classes.container}>
                    <Grid item>
                        {item.enabled ? <span className={`${classes.chip} ${classes.active}`}>Active</span> : <span className={`${classes.chip} ${classes.inactive}`}>Inactive</span>}
                    </Grid>
                    <Grid item className={classes.date}>
                        Le {new Date(item.created_at).toLocaleDateString()}
                    </Grid>
                </Grid>
                <Grid container className={classes.container}>
                    <Grid item className={classes.title}>{item.title}</Grid>
                    <Grid item className={classes.creator}>{item.creator}</Grid>
                </Grid>
                <Grid container spacing={1} className={classes.KpiContainer}>
                    <Grid item>
                        {item.with_notification ? <Box className={`${classes.chip} ${classes.withBorder}`}><i className="fas fa-bell" /></Box> : <Box className={`${classes.chip} ${classes.withBorder}`}><i className="fas fa-bell-slash" /></Box>}
                    </Grid>
                    <Grid item>
                        <Box className={`${classes.chip} ${classes.withBorder}`}>
                            {item.nb_views} vue{item.nb_views > 1 && 's'}
                        </Box>
                    </Grid>
                    <Grid item>
                        <Box className={`${classes.chip} ${classes.withBorder}`}>
                            {item.nb_detail_views} vue{item.nb_detail_views > 1 && 's'} détaillée{item.nb_detail_views > 1 && 's'}
                        </Box>
                    </Grid>
                    <Grid item>
                        <Box className={`${classes.chip} ${classes.withBorder}`}>
                            {item.nb_ripostes} riposte{item.nb_ripostes > 1 && 's'}
                        </Box>
                    </Grid>
                </Grid>
                <Grid container justifyContent="space-between" className={classes.buttonContainer}>
                    <Grid item>
                        <Button
                            className={classes.editButton}
                            onClick={() => handleClickOpen(item.uuid)}
                        >Éditer
                        </Button>
                    </Grid>
                    <Grid item>
                        <DotMenu handleActiveItem={handleActiveItem} item={item} />
                    </Grid>
                </Grid>
            </Paper>
        </Grid>
    );
};

export default Card;

Card.propTypes = {
    item: PropTypes.object.isRequired,
    handleClickOpen: PropTypes.func.isRequired,
    handleActiveItem: PropTypes.func.isRequired,
};
