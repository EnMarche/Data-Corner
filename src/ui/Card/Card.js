import PropTypes from 'prop-types'
import { Paper } from '@mui/material'
import { styled } from '@mui/system'

import { VerticalContainer } from 'components/shared/styled'

export const Root = styled(Paper)(
  ({ theme }) => `
  padding: ${theme.spacing(0, 2, 2, 2)};
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  border-radius: 8px;
  box-shadow: 0px 0px 5px rgba(29, 33, 79, 0.03);
  &:hover {
    box-shadow: 0px 4px 10px rgba(29, 33, 79, 0.06);
  }
`
)

const UICard = ({
  header,
  content,
  actions,
  rootProps = {},
  headerProps = {},
  contentProps = {},
  actionsProps = {},
}) => (
  <Root {...rootProps} data-cy="ui-card">
    {header && <VerticalContainer {...headerProps}>{header}</VerticalContainer>}
    {content && <VerticalContainer {...contentProps}>{content}</VerticalContainer>}
    {actions && <VerticalContainer {...actionsProps}>{actions}</VerticalContainer>}
  </Root>
)

UICard.propTypes = {
  header: PropTypes.node,
  content: PropTypes.node,
  actions: PropTypes.node,
  rootProps: PropTypes.object,
  headerProps: PropTypes.object,
  contentProps: PropTypes.object,
  actionsProps: PropTypes.object,
}

export default UICard
