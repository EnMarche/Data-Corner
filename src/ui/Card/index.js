export { default } from './Card'

export { Title } from './Title/Title'
export { Chip } from './Chip/Chip'
export { CtaButton } from './CtaButton/CtaButton'
