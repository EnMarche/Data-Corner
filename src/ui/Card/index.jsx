export { default } from './Card'

export { Title } from './Title/Title'
export { UIChip } from './Chip/Chip'
export { CtaButton } from './CtaButton/CtaButton'
