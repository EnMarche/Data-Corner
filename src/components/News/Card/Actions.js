import { CtaButton } from 'ui/Card'
import { styled } from '@mui/system'
import PropTypes from 'prop-types'
import DotsMenu from 'ui/Card/Menu/DotsMenu'

const HorizontalContainer = styled('div')`
  display: flex;
  flex: 1;
  align-items: flex-end;
  justify-content: space-between;
`

const messages = {
  see: 'Voir',
  published: 'Publier',
  unPublished: 'DÃ©publier',
}

const Actions = ({ status, toggleStatus, onView }) => (
  <HorizontalContainer>
    <CtaButton
      onClick={onView}
      sx={{
        color: 'orange500',
        '&:hover': {
          bgcolor: 'newsBackground',
        },
      }}
    >
      {messages.see}
    </CtaButton>
    <DotsMenu>
      <DotsMenu.Item onClick={toggleStatus}>{status ? messages.unPublished : messages.published}</DotsMenu.Item>
    </DotsMenu>
  </HorizontalContainer>
)

export default Actions

Actions.propTypes = {
  status: PropTypes.bool.isRequired,
  toggleStatus: PropTypes.func.isRequired,
  onView: PropTypes.func.isRequired,
}
