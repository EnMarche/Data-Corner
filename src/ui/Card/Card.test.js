import { render } from '@testing-library/react'
import Card from './Card'

jest.mock('@material-ui/core', () => ({
  Grid: ({ item, children, ...props }) => <div {...props}>{children}</div>,
  makeStyles: () => () => ({ root: 'root' }),
}))

describe('Card', () => {
  it('displays card', () => {
    const { container } = render(
      <Card rootClasses="fooClass" xs={0} sm={1} md={2} lg={3} xl={4}>
        Foo
      </Card>
    )

    expect(container).toMatchSnapshot()
  })
})
