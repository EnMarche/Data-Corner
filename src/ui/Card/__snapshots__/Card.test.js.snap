// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card displays a Card with actions 1`] = `
<div>
  <div
    class="mui-card-mock"
  >
    <div
      class="mui-cardcontent-mock"
    />
    <div
      class="mui-cardactions-mock"
    >
      actions
    </div>
  </div>
</div>
`;

exports[`Card displays a Card with content 1`] = `
<div>
  <div
    class="mui-card-mock"
  >
    <div
      class="mui-cardcontent-mock"
    >
      content
    </div>
  </div>
</div>
`;

exports[`Card displays a Card with headers 1`] = `
<div>
  <div
    class="mui-box-mock"
  >
    <div
      class="mui-box-mock"
    >
      header
      subtitle
    </div>
    <div
      class="mui-box-mock"
    />
  </div>
</div>
`;
